fix(results): count completed_samples from the sample log, not scored samples (#4602) (#4604)
UKGovernmentBEIS/inspect_ai: CHANGELOG.md

- Bugfix: Transcript timelines no longer hide workflow `generate()` calls with differing system prompts as utility agents; utility classification now requires a tool-calling agent loop.
- Bugfix: JSON, CSV, and Hugging Face dataset loaders now return an empty dataset when passed `limit=0` instead of returning all samples.
- Bugfix: Parameterized score reducers (e.g. `at_least(2)`, `pass_at(2)`, `pass_k(2)`) now round-trip through the registry instead of raising `LookupError` when restored from a log.

> - Bugfix: `EvalResults.completed_samples` is now taken from the sample log (samples whose `error` is `None`) rather than the number of scored samples, so a score-on-error sample no longer changes the reported completion count based on where the raising scorer sits in the scorer list. Applies to both the normal and cancelled/aborted result paths. (#4602)

Reviewer: Nit: our changelog entries aim for ~25 words. Suggested trim:

```suggestion
- Bugfix: `EvalResults.completed_samples` now counts samples that completed without error rather than scored samples, so score-on-error samples no longer inflate it. (#4602)
```
